### PR TITLE
Replace project-proprietary Mingw build solution by rake-compiler-dock.

### PR DESCRIPTION
**Pull request by larskanis**
[Rake-compiler-dock](https://github.com/rake-compiler/rake-compiler-dock) is a more common build tool and environment for Windows RubyInstaller compatible binary gems. It supports ruby 1.8 to 2.2 and Windows, Linux and OS-X as the host OS. It runs on top of docker as well.

This fixes https://github.com/msgpack/msgpack-ruby/issues/73

This also updates rake-compiler to the latest version.

I tested the build process on Windows and Linux and I tested the resulting gems on Windows-7 with ruby-1.9.3-x86, ruby-2.0.0-x86, ruby-2.2.2-x86 and ruby-2.2.2-x64, so far.

### CODE REVIEW

File: ext/msgpack/extconf.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@
 have_func("rb_sym2str", ["ruby.h"])
 have_func("rb_str_intern", ["ruby.h"])
 
-unless RUBY_PLATFORM.include? 'mswin'
+unless RUBY_PLATFORM =~ /mswin|mingw/
```
Comment (larskanis) [inline ext/msgpack/extconf.rb]: This addresses issue https://github.com/msgpack/msgpack-ruby/issues/29

**Comment by cosmo0920** [comment]
Could you resolve merge conflicts?

**Comment by larskanis** [comment]
@cosmo0920 Yes, it's up-to-date now.

**Comment by tagomoris** [comment]
Merged. Great improvement. Thank you!

**Comment by larskanis** [comment]
Nice that it worked for you!

**Pull request merged**